### DISCUSSION TITLE
Gitea pubkey auth behind sshpiper?

### DISCUSSION THREAD

**Discussion by pwFoo** [tg123/sshpiper#99] [Q&A]
Hi @tg123,

I use sshpiper as sftp / ssh reverse proxy and it works fine! Thinking about pubkey instead of password and how to setup this the easiest way...  Looks like some steps need to be done like generate, place key pair at backend and sshpiper?

Gitea have a interface for users to place there pubkey, but I think that can't work and can't be automated or done by the user?

Do you know anyone who realized that use case?

**Comment by pwFoo** [comment]
Discussed also here: https://github.com/tg123/sshpiper/issues/48

But looks like there was no solution?
Maybe change the port and go arround sshpiper with a custom port?

**Comment by tg123** [comment]
you can have sshpiper to verify the pub key in gitea and you have to add another trust key to gitea

user privX -> sshpiper (verify pubkey of privX, map to privY ) -> gitea (verify pubkey of privY)

2 things to think:
 how to load pubkey of privX added by user in gitea, read gitea db?
 how to make gitea trust sshpiper by adding pubkey of privY to gitea, write to gitea db?
